packages/rest/src/__tests__/acceptance/routing/routing.acceptance.ts

```diff
@@ -361,30 +356,80 @@ describe('Routing', () => {
       });
   });
 
-  it('binds the current controller', async () => {
-    const app = givenAnApplication();
-    const server = await givenAServer(app);
-    const spec = anOpenApiSpec()
-      .withOperationReturningString('get', '/name', 'checkController')
-      .build();
+  describe('current controller', () => {
+    let app: Application;
+    let server: RestServer;
+    let controllerClass: Constructor<ControllerInstance>;
 
-    @api(spec)
-    class GetCurrentController {
-      async checkController(
-        @inject('controllers.GetCurrentController') inst: GetCurrentController,
-      ): Promise<object> {
-        return {
-          result: this === inst,
-        };
+    beforeEach(setupController);
+
+    it('binds current controller resolved from a transient binding', async () => {
+      givenControllerInApp(app, controllerClass);
+
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 1,
+          isSingleton: false,
+          result: true,
+        });
+
+      // Make a second call
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 1, // The count is still 1 as it's from a new instance
+          isSingleton: false,
+          result: true,
+        });
+    });
+
+    it('binds current controller resolved from a singleton binding', async () => {
+      givenControllerAsSingletonInApp(app, controllerClass);
+
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 1,
+          isSingleton: true,
+          result: true,
+        });
+
+      // Make a second call
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 2, // The count increases as the controller is singleton
+          isSingleton: true,
+          result: true,
+        });
+    });
+
+    async function setupController() {
+      app = givenAnApplication();
+      server = await givenAServer(app);
```

Comment:
The helper says `setupController`, but under the hood it's also setting up `app` and `server`. I find that very confusing!

I think you should split this helper into two parts:

- `setupServerInApp`
- `setupController`

We use `given` instead of `setup` prefix in other places, `givenServerInApp` and `givenController` are probably more consistent names.

Reply:
> The helper says setupController, but under the hood it's also setting up app and server. I find that very confusing!
> 
> I think you should split this helper into two parts

☝️